gpu/command_buffer/client/cmd_buffer_helper.cc

```diff
@@ -197,6 +213,17 @@ void CommandBufferHelper::OrderingBarrier() {
     put_ = 0;
 
   if (HaveRingBuffer()) {
+#if defined(CASTANETS)
+    // If ring_buffer_ overflowed.
+    if (put_ < last_ordering_barrier_put_) {
+      size_t tail_size = ring_buffer_size_ - (last_ordering_barrier_put_ * kCommandBufferEntrySize);
+      mojo::SyncSharedMemoryHandle(ring_buffer_->backing()->GetGUID(), last_ordering_barrier_put_ * kCommandBufferEntrySize, tail_size);
+      mojo::SyncSharedMemoryHandle(ring_buffer_->backing()->GetGUID(), 0, put_ * kCommandBufferEntrySize);
+    } else {
+      size_t size = (put_ - last_ordering_barrier_put_) * kCommandBufferEntrySize;
+      mojo::SyncSharedMemoryHandle(ring_buffer_->backing()->GetGUID(), last_ordering_barrier_put_ * kCommandBufferEntrySize, size);
+    }
+#endif
```

Comment:
All the same with the above code. Add a new method?

Reply:
Only sync in renderer process.
if (base::CommandLine::ForCurrentProcess()->GetSwitchValueASCII("type") == "renderer")

Reply:
I updated a commit.